01_api-funct-iter.Rmd

```diff
@@ -0,0 +1,286 @@
+my_other_sentence <- "pints of ice cream a day"
+
+paste0(my_sentence, " ", 08, " ", my_other_sentence, "!")
+```
+
+So, if we make `year = 2021` in our `parkwide_visitation()` function, the year object becomes the number 2021, which makes the `paste0()` output "<https://irmaservices.nps.gov/v3/rest/stats/total/2021>", which subsequently pulls data for 2021. In other words, we can now pull visitation data for any year with just one line of code!
+
+```{r}
+pull_2018 <- parkwide_visitation(year = 2018)
+
+pull_1980 <- parkwide_visitation(year = 1980)
+
+pull_1992 <- parkwide_visitation(year = 1992)
+
+# ... and so on!
+```
+
+### Exercise #3 {style="color: maroon"}
+
+**Create a function called `unit_visitation()` that pulls park-specific visitation data for any park, across any time frame. For a list of all park codes, download [this spreadsheet](https://www.nps.gov/aboutus/foia/upload/NPS-Unit-List.xlsx). (Hint 1: functions can have multiple arguments. For this step, you will want arguments representing the start and end month and year, and park unit). Hint 2: Exercise 2 should be used as a starting point for making this function.)**
+
+```{r}
+
+```
+
+### Exercise #4 {style="color: maroon"}
+
+**Using `unit_visitation()`, pull in visitation data for Rocky Mountain National Park (ROMO), Everglades National Park (EVER), and Theodore Roosevelt National Park (THRO) from November 1992 through December 2021.**
+
+```{r}
+
+```
+
+## Function Defaults
+
+Look at the code that you just wrote; writing out all of those unchanging date arguments still feels repetitive, right? When developing functions, there is an option for setting default values for arguments so that you don't necessarily have to write all of them out every time you run it in the future. But, the option still exists within the function to make changes when necessary. For example, let's tweak our `parkwide_visitaion()` function to have the default year be 2021:
+
+```{r}
+parkwide_visitation <- function(year = "2021") {
+
+raw_data <- httr::GET(url = paste0("https://irmaservices.nps.gov/v3/rest/stats/total/", year))
+
+# convert content to text
+extracted_data <- httr::content(raw_data, as = "text", encoding = "UTF-8") 
+
+# parse text from JSON to data frame
+final_data <- jsonlite::fromJSON(extracted_data)
+
+return(final_data)
+
+}
+
+parkwide_visitation()
+```
+
+Because the default year is 2021, you don't have to write it out explicitly in the function (so long as that's the year you're interested in). But, you still have the option of changing the year to something else:
+
+```{r}
+parkwide_visitation(year = "1992")
+```
+
+### Exercise #5 {style="color: maroon"}
+
+**Default our `unit_visitation()` function's arguments related to the starting and ending months to January and December, respectively. This way, we are automatically pulling in data for entire years. Rerun the function for ROMO, EVER, and THRO for the 1980-2021 time period to make sure it works properly.**
```

Comment:
I think a typo here?

Reply:
where?

Reply:
The first sentence

Reply:
Oh wait, is 'Default' a verb here? Just going to suggest a re-wording to make more clear